UPDATE Jenkinsfile for parallel stage execution

robnester-rh opened the pull request:
Updated the Jenkinsfile and associated library files to allow testing
stages to run in parallel.

johnbieren commented on Jenkinsfile:
```diff
@@ -322,107 +322,119 @@ podTemplate(name: podName,
+                                pipelineUtils.sendMessageWithAudit(messageFields['properties'], messageFields['content'], msgAuditFile, fedmsgRetryCount)
+                            }
+                        }
+                    }, functionalj-tests {
```
typo

johnbieren commented on Jenkinsfile:
```diff
@@ -322,107 +322,119 @@ podTemplate(name: podName,
+                                pipelineUtils.executeInContainer(currentStage, "ostree-boot-image", "/home/ostree-boot-image.sh")
+
+                                // Set our message topic, properties, and content
+                                messageFields = pipelineUtils.setMessageFields("package.test.functional.queued")
```
Gotta move all the .queued messages to right before parallel step

johnbieren reviewed (approved):
lgtm

scoheb commented:
Lgtm

robnester-rh commented:
[test]

robnester-rh commented:
[test]

robnester-rh committed:
UPDATE Jenkinsfile to surround stages with withEnv

Updated the Jenkinsfile to surround stages with withEnv as a first step
to the eventual parallel run of some stages.

Updated the setStageEnvVars() method to return an ArrayList of strings
in the form "k=v" which can be used with the withEnv() call.
* Example:
* withEnv(pipelineUtils.setStageEnvVars(env.currentStage)){}

Updated the executeInContainer() method so that it will take an optional
argument (called vars), which is an ArrayList of "k=v" strings. Items in
this ArrayList will be added to the ArrayList returned from calling
env.getEnvironment(), already in the executeInContainer() method. This
avoids the need to set env vars repeatedly in the Jenkinsfile (such as
for rsync_paths, rsync_from, rsync_to)

Updated a few instances of variable interpolation to use more
"groovy-esque" variable interpolation in strings, instead of more
"java-esque" string concatenation.

robnester-rh committed:
Merge branch 'master' of github.com:CentOS-PaaS-SIG/ci-pipeline into parallel_stage_execution

robnester-rh commented:
Just a note, this latest push is meant for testing purposes; no need to review code at the moment.

scoheb commented on Jenkinsfile:
```diff
@@ -90,62 +90,62 @@ podTemplate(name: podName,
         containers: [
                 // This adds the custom slave container to the pod. Must be first with name 'jnlp'
                 containerTemplate(name: 'jnlp',
-                        image: DOCKER_REPO_URL + '/' + OPENSHIFT_NAMESPACE + '/jenkins-continuous-infra-slave:' + SLAVE_TAG,
```
you do not like my distinct touch here? ;)

robnester-rh committed:
Merge branch 'master' of github.com:CentOS-PaaS-SIG/ci-pipeline into parallel_stage_execution

robnester-rh committed:
Merge branch 'master' of github.com:CentOS-PaaS-SIG/ci-pipeline into parallel_stage_execution

robnester-rh committed:
UPDATE return from setStageEnvVar()

Updated setStageEnvVar to return a list compatible with the withEnv()
Jenkins API call.

robnester-rh committed:
UPDATE return for setStageEnvVars()

Added null object protection for setStageEnvVars

robnester-rh committed:
FIX typo in Jenkinsfile

robnester-rh committed:
ADD package to envVars for atomic-host-tests

robnester-rh commented:
[test]

robnester-rh committed:
ADD parallel for testing stages

robnester-rh commented:
[test]

robnester-rh commented:
[test]

robnester-rh committed:
REMOVE stages in parallel execution

Removed stages in parallel execution. Wrapped parallel execution in a
single 'validation' stage.

robnester-rh commented:
[test]

robnester-rh committed:
Debugging

robnester-rh commented:
[test]

robnester-rh commented:
[test]

robnester-rh commented:
[test]

robnester-rh commented:
[test]
